docs/features/network-security-controls/admin-network-policy.md

```diff
@@ -1,303 +1,1351 @@
 
-* They are not cluster-scoped and hence its hard to define a network of policies that spawns across namespaces
-* They cannot be created before the namespace is created (kapi server expects the namespace to be created first); thus they cannot satisfy the requirements where admins may want policies in the cluster to be in place before the workloads are even created.
-* The design of NetworkPolicy API is implicit which means the deny is already set in place when we create a policy and then we are expected to do an allowList of rules in the policy. Network administrators prefer to have the power of defining what exactly to deny and allow instead of this implicit model.
+* AdminNetworkPolicy API and
+* BaselineAdminNetworkPolicy API
 
-Sample API:
+which are shipped by the [sig-network-policy-api](https://network-policy-api.sigs.k8s.io/)
+working group as out-of-tree Kubernetes APIs. Both these APIs can be used to define
+cluster-wide admin policies in order to secure your cluster’s network traffic. These APIs
+are currently in the v1alpha1 version and subject to change based on upstream developments
+before becoming stable v1.
 
+## Motivation
+
+NetworkPolicies in Kubernetes were created for developers to be able to secure their
+applications. Hence they are under the control of potentially untrusted application authors.
+They are namespace scoped and are not suitable for solving the above user stories for cluster
+administrators because:
+
+* They are not cluster-scoped and hence its hard to define a network of policies
+  that spans across namespaces - you'd have to duplicate the same policy over and over
+  in each namespace
+* They cannot be created before the namespace is created (KAPI server expects the namespace
+  to be created first since it's namespace scoped); thus they cannot satisfy the requirements
+  where admins may want the policies to be applicable to arbitrary future
+  namespaces that could get created in the cluster.
+
+Another motivating was to fix some of the usability aspects of the NetworkPolicy API.
+The design of NetworkPolicy API is implicit which means the deny is already set in
+place when we create a policy and then we are expected to do an allowList of rules
+in the policy. Then we make other exceptions like `except` blocks in `ipBlock` construct
+which is an explicit what not to allow within the allowList. Network administrators prefer
+to have the power of defining what exactly to deny and allow instead of this implicit model.
+
+### User-Stories/Use-Cases
+
+Check out the original [User Stories](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/2091-admin-network-policy#user-stories)
+for AdminNetworkPolicy. However note that that proposal is stale and it's better to
+follow the new developments via the [Network Policy Enhancement Proposals](https://network-policy-api.sigs.k8s.io/enhancements/).
+Let us look at some practical use cases of when to use admin network policies
```

Comment:
Do you think we need the use cases here? Feels like it will become stale very soon, and we can just link to the upstream docs. (and do a better job upstream if the docs there are not good enough, but at least we will do that in one place)
Also, may be better to link user stories to https://network-policy-api.sigs.k8s.io/user-stories/ instead of the original KEP?

Reply:
yeah I have changed the link to link to upstream website user stories.